### ISSUE TITLE
ToCells (maybe) doesn't terminate

### ISSUE THREAD

**Issue by JayKickliter** [HydroniumLabs/h3o#21]
I'm attempting to polyfill a a 3º×3º box at res0. It doesn't look like this will ever terminate, even with new `ContainmentMode::Covers`. Here's reproduction code:

```rust
use geo::{geometry::GeometryCollection, polygon};
use h3o::{
    geom::{ContainmentMode, PolyfillConfig, Polygon, ToCells},
    CellIndex, Resolution,
};

fn main() {
    let geo_poly = polygon![
        (x: 156.0, y: 6.0),
        (x: 156.0, y: 3.0),
        (x: 159.0, y: 3.0),
        (x: 159.0, y: 6.0),
        (x: 156.0, y: 6.0),
    ];
    let geo_json =
        geojson::FeatureCollection::from(&GeometryCollection::from(vec![geo_poly.clone()]));
    println!("{geo_json}");

    let h3o_poly = Polygon::from_degrees(geo_poly).unwrap();

    let polyfill: Vec<CellIndex> = h3o_poly
        .to_cells(PolyfillConfig::new(Resolution::Zero).containment_mode(ContainmentMode::Covers))
        .collect();

    println!("{polyfill:?}");
}
```

And a geosjon of the polygon:

```json
{"features":[{"geometry":{"coordinates":[[[156.0,6.0],[156.0,3.0],[159.0,3.0],[159.0,6.0],[156.0,6.0]]],"type":"Polygon"},"properties":{},"type":"Feature"}],"type":"FeatureCollection"}
```

I know I can workaround this by generating all res0 cells and geometry hit testing them against this rect, but this'll be unfeasible if the caller switches to a higher resolution.

**Comment by grim7reaper** [comment]
Thanks for the report and the test case.

This trips a `debug_assert` in debug mode, I look into it.

**Comment by grim7reaper** [comment]
Fixed in v0.5.1
